### DISCUSSION TITLE
[OKD 4.12] Bootstrap failed

### DISCUSSION THREAD

**Discussion by ObieBent** [okd-project/okd#1671] [General]
**Describe the bug**
Bootstrap cannot finish the installation. 

**Version**
4.12.0-0.okd-2023-04-16-041331
UPI

**How reproducible**

100%

**Log bundle**

[log-bundle-20230726101922.tar.gz](https://github.com/okd-project/okd/files/12170311/log-bundle-20230726101922.tar.gz)

Hello guys, 
I am currently trying install OKD 4.12 cluster through Terraform/KVM. 
DNS, HAProxy, DHCP are up and running. 

When I try to boot the bootstrap, it stuck over and over and failed to bring up the API. 

DEBUG Still waiting for the Kubernetes API: Get "https://api.cluster-okd.buzz.lab:6443/version": EOF
DEBUG Still waiting for the Kubernetes API: Get "https://api.cluster-okd.buzz.lab:6443/version": EOF
ERROR Bootstrap failed to complete: Get "https://api.cluster-okd.buzz.lab:6443/version": EOF
ERROR Failed waiting for Kubernetes API. This error usually happens when there is a problem on the bootstrap host that prevents creating a temporary control plane.

I'm using this fedora-coreos-37.20221127.3.0-qemu.x86_64.qcow2.xz image. 

Best regards

**Comment by vrutkovs** [comment]
Masters didn't boot or couldn't reach bootstrap node - master Ignition requests were not recorded in machine-config-server machine. Check if these have boot, got DHCP address and can reach bootstrap node

**Reply by ObieBent** [reply]
The masters booted and got successfully DHCP address. 
My bad, I have commented out the statement related to bootstrap server in HAProxy configuration :( 
The API is up ! Thanks for the insights. 

I think you can close this issue. 

Regards
